airflow/api/client/local_client.py

```diff
@@ -36,18 +38,20 @@ def delete_dag(self, dag_id):
         return f"Removed {count} record(s)"
 
     def get_pool(self, name):
-        the_pool = pool.get_pool(name=name)
+        the_pool = Pool.get_pool(pool_name=name)
+        if not the_pool:
+            raise PoolNotFound(f"Pool {name} not found")
```

Comment:
Previously, the `get_pool` experimental API raises PoolNotFound when the pool does not exist. Since I have moved it to the Pool model, I don't want it to raise hence raising not found here

Reply:
Looking at where this function is being called, I wonder if we should just get rid of this exception altogether and just return `Optional[Pool]`.